pulsar-common/src/main/java/org/apache/pulsar/common/util/collections/GrowablePriorityLongPairQueue.java

```diff
@@ -204,7 +207,7 @@ private LongPair removeAtWithoutLock(int index) {
             LongPair item = new LongPair(data[index], data[index + 1]);
             data[index] = EmptyItem;
             data[index + 1] = EmptyItem;
-            --this.size;
```

Comment:
This is already protected by a write lock. No need to use updater. The volatile is only used to ensure readers are reading the last value of the size.